sstables/sstables.hh

```diff
@@ -953,6 +957,13 @@ public:
     // Returns a read-only file for all existing components of the sstable
     future<std::unordered_map<component_type, file>> readable_file_for_all_components() const;
 
+    struct lesser_reclaimed_memory {
+        // comparator class to be used by the _reclaimed set in sstables manager
+        bool operator()(const sstable& sst1, const sstable& sst2) const {
+            return sst1.total_memory_reclaimed() < sst2.total_memory_reclaimed();
+        }
```

Comment:
Is this sufficient? It's not a total order.

Reply:
Yes, the boost [doc](https://www.boost.org/doc/libs/1_83_0/doc/html/intrusive/set_multiset.html) for intrusive set says that the `comparison functor must induce a strict weak ordering`. So, this should be sufficient.